src/compiler/transformers/declarations/diagnostics.ts

```diff
@@ -0,0 +1,469 @@
+/* @internal */
+namespace ts.declarations {

```

Comment:
The file is for organization. They have no coupling to the transformer, and I'm not interested in almost doubling the length of the transformer's file just for diagnostic selection. (If it was simple to extract, I'd have each if the visitors separate, too.) They don't really need to be in a sub-namespace, I used one mostly because at first I wasn't sure how many methods I'd use - in the end it's only around two or three.

Reply:
That's what _compiler/transformers/utilities.ts_ is for.

Reply:
I don't think so. These aren't general transformer utilities, and a `utilities` file is more of an antipattern than anything else - it's a dumping ground for code with no more specific home, and these aren't some small five-line helper. These are diagnostics for the declaration transform. These are 1000+ lines of code. They are not needed in any other context than if you're investigating the diagnostics of the declaration emitter. I think they should stay on their own.